### PR TITLE
Update Covid configs

### PR DESCRIPTION
**Pull request by Shruthi42**

### CODE REVIEW

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 1]
```diff
@@ -234,24 +246,38 @@ def get_output_csv_path(mode: ModelExecutionMode) -> Path:
             p = get_best_epoch_results_path(mode=mode, model_proc=model_proc)
             return self.outputs_folder / p / MODEL_OUTPUT_CSV
 
+        def get_labels_and_predictions(df: pd.DataFrame) -> pd.DataFrame:
+            labels = []
+            predictions = []
+            for i, target in enumerate(self.target_names):
+                predictions.append(df[df[LoggingColumns.Hue.value] == target][LoggingColumns.ModelOutput.value].item())
+                labels.append(df[df[LoggingColumns.Hue.value] == target][LoggingColumns.Label.value])
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:254]: I'd define a variable `target_df = df[df[LoggingColumns.Hue.value] == target]` to make it a bit more readable.

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 2]
```diff
@@ -234,24 +246,38 @@ def get_output_csv_path(mode: ModelExecutionMode) -> Path:
             p = get_best_epoch_results_path(mode=mode, model_proc=model_proc)
             return self.outputs_folder / p / MODEL_OUTPUT_CSV
 
+        def get_labels_and_predictions(df: pd.DataFrame) -> pd.DataFrame:
+            labels = []
+            predictions = []
+            for i, target in enumerate(self.target_names):
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:252]: Does this need enumeration? `i` is unused.

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 3]
```diff
@@ -192,39 +191,52 @@ def _get_ssl_checkpoint_path(self) -> Path:
...
+                LoggingColumns.DataSplit.value: data_split.value
+            })
 
     def generate_custom_report(self, report_dir: Path, model_proc: ModelProcessing) -> Path:
         """
-        Generate a custom report for the CovidDataset Hierarchical model. At the moment, this report will read the
-        file model_output.csv generated for the training, validation or test sets and compute a 4 class accuracy
-        and confusion matrix based on this.
+        Generate a custom report for the Covid model. This report will read the file model_output.csv generated for
+        the training, validation or test sets and compute a 4 class accuracy and confusion matrix based on this.
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:239]: There doesn't seem to be a confusion matrix anymore; we should update the docstring.

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 4]
```diff
@@ -234,24 +246,38 @@ def get_output_csv_path(mode: ModelExecutionMode) -> Path:
...
+            return pd.DataFrame.from_dict({LoggingColumns.Patient.value: [df.iloc[0][LoggingColumns.Patient.value]],
+                                           LoggingColumns.ModelOutput.value: [np.argmax(predictions)],
+                                           LoggingColumns.Label.value: [np.argmax(labels)]})
+
+        def get_accuracy(df: pd.DataFrame) -> float:
+            df = df.groupby(LoggingColumns.Patient.value, as_index=False).apply(get_labels_and_predictions).reset_index(
+                drop=True)
+            df["tp+tn"] = df.apply(
+                lambda x: 1 if x[LoggingColumns.ModelOutput.value] == x[LoggingColumns.Label.value] else 0, axis=1)
+            return np.sum(df["tp+tn"].values) / len(df)
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:265]: Boolean `pd.Series` support arithmetic operations, so you can just do e.g. `(df[A] == df[B]).mean()`.

File: InnerEye/ML/configs/classification/CovidModel.py [hunk 5]
```diff
@@ -234,24 +246,38 @@ def get_output_csv_path(mode: ModelExecutionMode) -> Path:
             p = get_best_epoch_results_path(mode=mode, model_proc=model_proc)
             return self.outputs_folder / p / MODEL_OUTPUT_CSV
 
+        def get_labels_and_predictions(df: pd.DataFrame) -> pd.DataFrame:
+            labels = []
+            predictions = []
+            for i, target in enumerate(self.target_names):
+                predictions.append(df[df[LoggingColumns.Hue.value] == target][LoggingColumns.ModelOutput.value].item())
```
Comment (dccastro) [inline InnerEye/ML/configs/classification/CovidModel.py:253]: Why are we calling `.item()`?

**Review by dccastro — CHANGES_REQUESTED** [review]

**Review by dumbledad — APPROVED** [review]

**Pull request merged**